### ISSUE TITLE
[BUG] Configure generates invalid Makefile for clang, if AMD runtime is installed

### ISSUE THREAD

**Issue by AngryLoki** [lsof-org/lsof#305]
**Describe the bug**
This issue was originally reported to https://bugs.gentoo.org/919253

When clang is used, and HIP runtime (amdgpu) is installed, build fails with:
```
>>> Compiling source in /var/tmp/portage/sys-process/lsof-4.99.0/work/lsof-4.99.0 ...

make -j4 DEBUG= all 
Makefile:1120: *** missing separator.  Stop.
```

The issue is caused by line https://github.com/lsof-org/lsof/blob/4.99.0/configure.ac#L559 , which attempts to extract clang version, while `clang -v` additionally outputs version of HIP runtime. This issue is similar to #68, but there it was clang/macos/nvidia and now it affects clang/linux/amdgpu.

```
$ clang -v 2>&1 | grep version
clang version 17.0.3
Found HIP installation: /usr/local, version 5.7.31921
```

It is very likely that as described in #68, Makefile will also fail with `Found CUDA installation: /usr/local/cuda, version 9.0` too

**To Reproduce**
1. Install hip runtime and clang. Try to build with clang
2. See error

**Expected behavior**
As clang outputs its version in the first line, and may output arbitrary information after that, it should be safe to take version from the first `version` in the output.

 - Kernel: Linux
 - OS: Gentoo
 - lsof Version: 4.99.0
 - Origin: lsof update during system upgrade

**Comment by AngryLoki** [comment]
Closed with fix in https://github.com/lsof-org/lsof/pull/306, thanks for merging

**Comment by jiegec** [comment]
Fixed in #306
